code/modules/mapping/reader.dm

```diff
@@ -60,25 +72,37 @@
 	placeOnTop = FALSE as num,
 	orientation = SOUTH as num,
 	annihilate_tiles = FALSE,
-	z_lower = -INFINITY as num,
-	z_upper = INFINITY as num
+	crop_relative_to_game_world = TRUE
 	)
-	var/datum/parsed_map/parsed = new(dmm_file, x_lower, x_upper, y_lower, y_upper, z_lower, z_upper, measureOnly)
+	var/datum/parsed_map/parsed = new(dmm_file, measureOnly = measureOnly)
 	if(parsed.bounds && !measureOnly)
 		parsed.load(x_offset, y_offset, z_offset, cropMap, no_changeturf, x_lower, x_upper, y_lower, y_upper, placeOnTop, orientation, annihilate_tiles)
 	return parsed
 
-/// Parse a map, possibly cropping it.
-//WHY THE HECK DO WE EVEN SUPPORT NEGATIVE COORDINATES, ALL IT IS IS A WASTE OF TIME AND CPU!!!???
-//DO NOT USE THIS TO TRIM MAPS UNLESS STRICTLY NEEDED! IT IS EXTREMELY EXPENSIVE TO DO SO!
+/**
+  * Parse a map, possibly cropping it.
+  * Do not use the crop function unless strictly necessary.
+  * WARNING: Crop function crops based on the tiles you'd see in the map editor. If you're planning to load it in in a different orientation later, you better have done the math.
+  * It's recommended that you do not crop using this at all.
+  */
 /datum/parsed_map/New(tfile, x_lower = -INFINITY, x_upper = INFINITY, y_lower = -INFINITY, y_upper = INFINITY, z_lower = -INFINITY, z_upper = INFINITY, measureOnly = FALSE)
+	_parse(tfile, x_lower, x_upper, y_lower, y_upper, z_lower, z_upper, measureOnly)
+
+/datum/parsed_map/proc/_parse(tfile, x_lower, x_upper, y_lower, y_upper, z_lower, z_upper, measureOnly)
```

Comment:
dmmregex and other things are statics, i'm pretty sure they can't collide but for the sake of performance no one should be trying to load/parse two different maps at the same time anyways.